### PR TITLE
Feat: Webauthn based authentication

### PR DESCRIPTION
**Pull request by mattbell87**
## Checklist
- [ ] I'm using the correct PHP Version (8.1 for current, 7.4 for legacy).
- [ ] I've added comments to any new methods I've created or where else relevant.
- [ ] I've replaced magic method usage on DbService classes with the getInstance() static method.
- [ ] I've written any documentation for new features or where else relevant in the docs [repo](https://github.com/2pisoftware/cmfive-docs).

## Changelog

refs:
issues:

## Other Information

Docs pull request:

### CODE REVIEW

File: system/modules/timelog/tests/acceptance/playwright/timelog.test.ts [hunk 1]
```diff
@@ -36,8 +36,8 @@ test("You can create a Timelog using Add Timelog" , async ({page}) => {
     
     const taskgroup = CmfiveHelper.randomID("taskgroup_");
     const taskgroupID = await TaskHelper.createTaskGroup(page, taskgroup, "Software Development", "OWNER", "OWNER", "OWNER");
-    await TaskHelper.addMemberToTaskgroup(page, taskgroup, taskgroupID, "admin admin", "OWNER");
-    await TaskHelper.setDefaultAssignee(page, taskgroup, taskgroupID, "admin admin");
+    await TaskHelper.addMemberToTaskgroup(page, taskgroup, taskgroupID, "Admin Admin", "OWNER");
```
Comment (adam-buckley) [inline system/modules/timelog/tests/acceptance/playwright/timelog.test.ts:39]: We'll need a better way of dictating what admin user should be used to run the tests but by default the test setup tool names it "admin admin" so this change will break tests in Github Actions - please revert

**Review by adam-buckley — CHANGES_REQUESTED** [review]

File: system/modules/timelog/tests/acceptance/playwright/timelog.test.ts (hunk 1 above)
Comment (mattbell87) [reply system/modules/timelog/tests/acceptance/playwright/timelog.test.ts:39]: Oops, these changes were not meant to be here anyway!

File: system/modules/admin/install/migrations/20240808181623-AdminMigration9izg7_TestMigration.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+class AdminMigration9izg7_TestMigration extends CmfiveMigration
+{
+    public function up()
```
Comment (adam-buckley) [inline system/modules/admin/install/migrations/20240808181623-AdminMigration9izg7_TestMigration.php:5]: This commit from the test system shouldn't be here

**Review by adam-buckley — CHANGES_REQUESTED** [review]

File: system/modules/admin/install/migrations/20240808181623-AdminMigration9izg7_TestMigration.php (hunk 2 above)
Comment (adam-buckley) [reply system/modules/admin/install/migrations/20240808181623-AdminMigration9izg7_TestMigration.php:5]: migration*

File: system/modules/auth/actions/ajax_register_authenticator.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    $user_id = Request::int("id");
+    if (empty($user_id)) {
+        $w->out((new JsonResponse())->setErrorResponse("Request data missing", null));
+        return;
+    }
+
+    // RP Entity i.e. the application
+    $rpEntity = PublicKeyCredentialRpEntity::create(
+        'CMfive', //Name
+        'cmfive.com',              //ID
```
Comment (adam-buckley) [inline system/modules/auth/actions/ajax_register_authenticator.php:23]: I know this is still draft but this name/domain should be configurable via cmfive Config

File: system/modules/auth/actions/ajax_register_authenticator.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        null                            //Icon
+    );
+
+    // User Entity
+    $userEntity = PublicKeyCredentialUserEntity::create(
+        'admin',                   //Name
+        $user_id, //ID
+        'admin admin',                          //Display name
```
Comment (adam-buckley) [inline system/modules/auth/actions/ajax_register_authenticator.php:31]: Same for the name

File: system/modules/auth/actions/ajax_register_authenticator.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        null                                    //Icon
+    );
+
+    // Challenge
+    $challenge = random_bytes(16);
+
+    $publicKeyCredentialCreationOptions =
+        PublicKeyCredentialCreationOptions::create(
+            $rpEntity,
```
Comment (adam-buckley) [inline system/modules/auth/actions/ajax_register_authenticator.php:40]: Do we also need to store this status in the database somewhere? From memory we do it for MFA

**Pull request closed without merging**